### PR TITLE
Paris - Niccoló & Laura

### PR DESCRIPTION
**Pull request by laurastromboni**
@nicrodemus

### CODE REVIEW

**Comment by laurastromboni** [comment]
We've not finished the 2 last questions yet

**Comment by laurastromboni** [comment]
Rest to do : Bonus

File: starter-code/src/functions-and-arrays.js [hunk 1]
```diff
@@ -74,6 +186,24 @@ var wordsCount = [
   'disobedience',
   'matter'
 ];
+
```
Comment (ta-web-paris) [inline starter-code/src/functions-and-arrays.js]: here is a corrected version of this part : 
```
function howManyTimes (wordsCount, wordTried){
  var searchResult = 0;
  if (wordsCount.length === 0){
    return false;
  }
  for (i=0; i<wordsCount.length; i++){
    if (wordsCount[i] === wordTried  ){ // returns true if the word we are looking is the only one on the array
      searchResult++ ;
    }
  }
  return searchResult;

}
```

You were ALMOST there :)

File: starter-code/src/functions-and-arrays.js [hunk 2]
```diff
@@ -74,6 +186,24 @@ var wordsCount = [
...
+    if (wordTried === wordsCount[i]){ // returns true if the word we are looking is the only one on the array
+      searchResult += ;
+    }
+
+  }
+
+}
+
+
 // Bonus Quest
```
Comment (ta-web-paris) [inline starter-code/src/functions-and-arrays.js]: And here is the correction for the matrix part : 
```
function greatestProduct(m) {
  var horizontal, vertical, diagonal;
  var greatProduct = 0;

  for (var i=0, nRows = m.length; i < nRows - 1; i++) {
    for (var j=0, nCols = m[0].length; j < nCols - 1; j++) {
      // 4 horizontal elements
      if (j < nCols-3) {
        horizontal = m[i][j] * m[i][j+1]   * m[i][j+2]   * m[i][j+3];
      }

      // 4 vertical elements
      if (i < nRows-3) {
        vertical   = m[i][j] * m[i+1][j]   * m[i+2][j]   * m[i+3][j];
      }

      // 4 Diagonal elements
      if (i < nRows-3 && j < nCols-3) {
        diagonal   = m[i][j] * m[i+1][j+1] * m[i+2][j+2] * m[i+3][j+3];
      }

      greatProduct = Math.max(greatProduct, horizontal, vertical, diagonal);
    }
  }

  return greatProduct;
}
```

But this one is quite tough!

File: starter-code/src/functions-and-arrays.js [hunk 3]
```diff
@@ -74,6 +186,24 @@ var wordsCount = [
...
 
 var matrix = [
```
Comment (ta-web-paris) [inline starter-code/src/functions-and-arrays.js]: Well done!
Your code is clean, you almost finished the full lab, which is encouraging! 
Keep up the good work :)

**Pull request closed without merging**